xde/src/xde.rs

```diff
@@ -1600,6 +1652,34 @@ unsafe fn xde_mc_tx_one(src_dev: &XdeDev, mut pkt: MsgBlk) -> *mut mblk_t {
             return ptr::null_mut();
         }
     };
+    let meoi_len = parsed_pkt.len() as u32;
+
+    let meta = parsed_pkt.meta();
+    let non_eth_payl_bytes = (&meta.inner_l3, &meta.inner_ulp).packet_length();
+
+    let (l4_flag, l4_ty) = match &meta.inner_ulp {
+        Some(ValidUlp::Tcp(_)) => {
+            (MacEtherOffloadFlags::L4INFO_SET, IpProtocol::TCP)
+        }
+        Some(ValidUlp::Udp(_)) => {
+            (MacEtherOffloadFlags::L4INFO_SET, IpProtocol::UDP)
+        }
+        _ => (MacEtherOffloadFlags::empty(), IpProtocol(0)),
+    };
+
+    let ulp_meoi = mac_ether_offload_info_t {
+        meoi_flags: MacEtherOffloadFlags::L2INFO_SET
+            | MacEtherOffloadFlags::L3INFO_SET
+            | l4_flag,
+        meoi_len,
+        meoi_l2hlen: meta.inner_eth.packet_length() as u8,
```

Comment:
taken literally this seems concerning, but i'm guessign `packet_lenght()` really means `header_lenght()`? The usize squashing is also a bit concerning.